docs/best_practices/tool_xml.rst

```diff
@@ -162,6 +162,12 @@ warning messages appear). See `planemo docs <https://planemo.readthedocs.io/en/l
 When tools contain output filters, tests should be included that verify
 this filtering occurs. See `planemo docs <https://planemo.readthedocs.io/en/latest/writing_how_do_i.html#test-output-filters-work>`__ for more information.
 
+Data parameters
+---------------
+
+If a (g)zipped data type is available (e.g. ``fasta.gz`` or ``fastq.gz``) then
+data parameters should accept both the zipped and the unzipped data type.
```

Comment:
```suggestion
If a compressed data type (e.g. ``fasta.gz`` or ``fastqsanger.bz2``) is
supported by the underlying tool, then data parameters should accept both the
compressed and the uncompressed data types.
```

Reply:
> supported by the underlying tool

this is to weak in my opinion. I would prefer to let the Galaxy tool unzip the data if the underlying tool does not support zipped files natively. Advantage is that this save a lot of space in user histories.

Reply:
Yes, but that's a galaxy change not a tool change

Reply:
I have something like this in mind: 

- https://github.com/galaxyproject/tools-iuc/pull/3542/files
- https://github.com/galaxyproject/tools-iuc/pull/3549/files
- https://github.com/galaxyproject/tools-iuc/pull/3548/files

Reply:
Sorry, I miss read that as preferring the Galaxy framework do this (which is/was my own view).

The examples make it clear you are advocating extending the tool wrapper to do decompression when the tool itself doesn't.

Reply:
> this is to weak in my opinion. I would prefer to let the Galaxy tool unzip the data if the underlying tool does not support zipped files natively. Advantage is that this save a lot of space in user histories.

I've updated my suggestion above accordingly.